standardnotes/mobile
# End-to-end tests

The following PR fixes the following minor bugs:

- 'Export Data' label doesn't change from 'Processing...' when the export process is intentionally interrupted/cancelled (https://github.com/standardnotes/mobile/issues/246)
- 'Sign out' button is still present after signing out

Additionally, end-to-end tests have been introduced into the codebase. Tests for the above changes have been included also.

## How to run end-to-end tests

1. Pull this branch (`e2e-testing`) and run `npm install`
1. Setup an Android emulator and launch it. The `detox` configuration object contains both `android.emu.debug` and `android.emu.release`, which both use a predefined AVD. In this case, a Google Pixel 2 XL was used. You can use a physical device too, but this is out of scope for the present PR :)
1. Run `npx react-native start` to start Metro Bundler.
1. Finally, run `detox test -c android.emu.debug` to start the Standard Notes app. Tests should start running.

![e2e-preview_2020-04-02](https://user-images.githubusercontent.com/5891646/78221099-c2c4a000-7490-11ea-9e9f-20bcf2da9a17.gif)

### Dependencies
- [detox](https://www.npmjs.com/package/detox)
- [mocha](https://www.npmjs.com/package/mocha)
- [faker](https://www.npmjs.com/package/faker)
